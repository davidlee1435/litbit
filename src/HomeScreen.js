import React from 'react';
import { Image, TouchableHighlight, Picker, TextInput, Button, StyleSheet, Text, View, FlatList } from 'react-native';
import * as firebase from 'firebase';

import AuthService from './service/AuthService.js';

const _ = require('lodash');

export default class HomeScreen extends React.Component {
  static navigationOptions = {
    title: "Home",
  };

  constructor() {
    super();
    this.authService = new AuthService();
    this.state = {
      cart: {
        1: 0,
        2: 0,
      },
      items: [
        {
          key: 1,
          title: 'Cups',
          imageUrl: require('./img/cupstxt1.png'),
          defaultQuantity: 10,
          pricePerDefaultQuantity: 2.99,
        },
        {
          key: 2,
          title: 'Balls',
          imageUrl: require('./img/ballstxt1.png'),
          defaultQuantity: 2,
          pricePerDefaultQuantity: 2.99,
        },
      ]
    }
  }

  checkoutCart() {
    var mergedCart = _.forEach(this.state.items, (item)=> {
      item.quantityOrdered = this.state.cart[item.key]
    })
    this.props.navigation.navigate('Confirm', {'user': this.props.screenProps.user, 'cart': mergedCart})
  }

  delivery() {
    this.props.navigation.navigate('Delivery')
  }

  clearCart() {
    this.setState(_.merge({}, this.state, {
      cart: _.mapValues(this.state.cart, () => 0)
    }))
  }

  onPress(item) {
    var count = this.state.cart[item.key];
    this.setState(_.merge({}, this.state, {
      cart: {
        [item.key]: count+1,
      }
    }))
  }

  render() {
    // const {items} = this.state.items;
    return (
      <View style={styles.container}>
        <Button
          style={styles.checkoutButton}
          onPress={() => {this.delivery()}}
          title="Delivery"
          color="#841584"
          accessibilityLabel="Delivery"
        />
        <Button
          style={styles.checkoutButton}
          onPress={() => {this.authService.signOut()}}
          title="Log out"
          color="#841584"
        />
        <FlatList
          contentContainerStyle={styles.feed}
          horizontal = {true}
          data = {this.state.items}
          renderItem={({item}) => {
            return (
              <View style={styles.itemButtonContainer}>
                <TouchableHighlight onPress={() => this.onPress(item)}>
                  <Image
                    style={styles.itemButton}
                    source={item.imageUrl}
                  />
                </TouchableHighlight>
              </View>
            )
          }}
        />
        <View style={styles.checkoutWrapper}>
          <View>
            <FlatList
              data = {this.state.items}
              renderItem={({item}) => {
                return (
                  <Text>
                    {item.title}: {this.state.cart[item.key] * item.defaultQuantity}
                  </Text>
                )
              }}
            />
          </View>

          <Button
            style={styles.checkoutButton}
            onPress={() => this.clearCart()}
            title="Clear Cart"
            color="#841584"
            accessibilityLabel="Clear Cart"
          />
          <Button
            style={styles.checkoutButton}
            disabled={_.sum(_.values(this.state.cart)) === 0}
            onPress={() => {this.checkoutCart()}}
            title="Checkout"
            color="#841584"
            accessibilityLabel="Checkout"
          />
        </View>
      </View>
	  )
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#ebebeb',
  },
  feed: {
    alignItems: 'center',
  },
  itemButtonContainer: {
    alignItems: 'center',
    backgroundColor: '#66b4f3',

  },
  itemButton: {
    height: 200,
    width:200,
  },
  checkoutButton: {
  },
  checkoutWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    padding: 15,
  },
  submit:{
    // marginRight:70,
    // marginLeft:70,
    // marginTop:10,
    paddingTop:10,
    paddingBottom:10,
    paddingRight:10,
    paddingLeft:10,
    backgroundColor:'grey',
    borderRadius:10,
    borderWidth: 1,
    borderColor: '#fff'
  },
  submitText:{
      color:'#fff',
      textAlign:'center',
  }
});
