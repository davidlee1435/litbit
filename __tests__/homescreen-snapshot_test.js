// Link.react-test.js
import React from 'react';
import HomeScreen from '../src/HomeScreen.js';
import renderer from 'react-test-renderer';

test('HomeScreen test', () => {
  const tree = renderer.create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();

  // tree.checkoutCart();
  // expect(tree).toMatchSnapshot();


  // // manually trigger the callback
  // tree.props.onMouseEnter();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // tree.props.onMouseLeave();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});







// import React from 'react'
// import HomeScreen from '../src/HomeScreen.js';
//  // checkoutCart() {
//  //    var mergedCart = _.forEach(this.state.items, (item)=> {
//  //      item.quantityOrdered = this.state.cart[item.key]
//  //    })
//  //    this.props.navigation.navigate('Confirm', {'user': this.props.screenProps.user, 'cart': mergedCart})
//  //  }

// test('renders correct navigation', (t) => {
//   const HomeScreen = new HomeScreen();
//   HomeScreen.nav = 'Confirm';
//   t.is(HomeScreen.nav, 'Confirm')
// })

// it('renders correctly', () => {
//   const tree = renderer.create(<App />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

// // test('user can be constructed with default parameters', t => {
// //   new HomeScreen({ platformType: 'cli', platformId: 1 });
// //   t.pass();
// // });