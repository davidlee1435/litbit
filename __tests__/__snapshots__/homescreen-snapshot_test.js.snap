// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen test 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ebebeb",
      "flex": 1,
    }
  }
>
  <View
    accessibilityComponentType="button"
    accessibilityLabel="Delivery"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#841584",
            },
          ]
        }
      >
        Delivery
      </Text>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
      }
    }
    data={
      Array [
        Object {
          "defaultQuantity": 10,
          "imageUrl": 1,
          "key": 1,
          "pricePerDefaultQuantity": 2.99,
          "title": "Cups",
        },
        Object {
          "defaultQuantity": 2,
          "imageUrl": 1,
          "key": 2,
          "pricePerDefaultQuantity": 2.99,
          "title": "Balls",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={undefined}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#66b4f3",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 200,
                  "width": 200,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#66b4f3",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 200,
                  "width": 200,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 15,
      }
    }
  >
    <View>
      <RCTScrollView
        data={
          Array [
            Object {
              "defaultQuantity": 10,
              "imageUrl": 1,
              "key": 1,
              "pricePerDefaultQuantity": 2.99,
              "title": "Cups",
            },
            Object {
              "defaultQuantity": 2,
              "imageUrl": 1,
              "key": 2,
              "pricePerDefaultQuantity": 2.99,
              "title": "Balls",
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        onViewableItemsChanged={undefined}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
            >
              Cups
              : 
              0
            </Text>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
            >
              Balls
              : 
              0
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Clear Cart"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
            ]
          }
        >
          Clear Cart
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Checkout"
      accessibilityTraits={
        Array [
          "button",
          "disabled",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Checkout
        </Text>
      </View>
    </View>
  </View>
</View>
`;
